src/components/WorkspaceSelectionDialog.js

```diff
@@ -2,55 +2,127 @@ import React, { Component } from 'react';
-import FormControl from '@material-ui/core/FormControl';
-import InputLabel from '@material-ui/core/InputLabel';
-import Select from '@material-ui/core/Select';
-import MenuItem from '@material-ui/core/MenuItem';
+import { List, ListItem, ListItemText } from '@material-ui/core';
 import PropTypes from 'prop-types';
+import { keys, chars } from '../lib/KeyHelper';
 
 /**
  */
 export class WorkspaceSelectionDialog extends Component {
+  /** */
+  constructor(props) {
+    super(props);
+    const { workspaceType } = this.props;
+    this.state = {
+      selected: workspaceType,
+    };
+
+    this.selectPreviousItem = this.selectPreviousItem.bind(this);
+    this.selectNextItem = this.selectNextItem.bind(this);
+    this.keyDownHandler = this.keyDownHandler.bind(this);
+    this.workspaceTypes = ['elastic', 'mosaic'];
+  }
+
+  /**
+   * Propagate workspace type selection into the global state
+   */
+  handleworkspaceTypeChange(workspaceType) {
+    const { handleClose, updateConfig } = this.props;
+
+    updateConfig({
+      workspace: {
+        type: workspaceType,
+      },
+    });
+    handleClose();
+  }
+
+  /** */
+  selectPreviousItem() {
+    const { selected } = this.state;
+    const selectedIndex = this.workspaceTypes.indexOf(selected);
+    const newIndex = selectedIndex > 0 ? selectedIndex - 1 : this.workspaceTypes.length - 1;
+    this.setState({ selected: this.workspaceTypes[newIndex] });
+  }
+
+  /** */
+  selectNextItem() {
+    const { selected } = this.state;
+    const selectedIndex = this.workspaceTypes.indexOf(selected);
+    const newIndex = selectedIndex + 1 < this.workspaceTypes.length ? selectedIndex + 1 : 0;
+    this.setState({ selected: this.workspaceTypes[newIndex] });
+  }
+
+  /** */
+  keyDownHandler(event) {
+    if (event.key === keys.up || event.which === chars.up) {
+      event.preventDefault();
+      return this.selectPreviousItem(event.target);
+    }
+    if (event.key === keys.down || event.which === chars.down) {
+      event.preventDefault();
+      return this.selectNextItem(event.target);
+    }
+    if (event.key === keys.enter || event.which === chars.enter) {
+      event.preventDefault();
+      const { selected } = this.state;
+      this.handleworkspaceTypeChange(selected);
+    }
+    return null;
+  }
+
   /**
    * render
    * @return
    */
   render() {
     const {
-      container, handleClose, open, children, workspaceType, updateConfig, t,
+      classes, container, handleClose, open, children, t,
     } = this.props;
     return (
-      <Dialog id="workspace-settings" container={container} open={open} onClose={handleClose}>
-        <DialogTitle id="form-dialog-title">{t('workspaceSelectionTitle')}</DialogTitle>
+      <Dialog
+        aria-labelledby="workspace-selection-dialog-title"
+        container={container}
+        id="workspace-settings"
+        onClose={handleClose}
+        onKeyDown={event => this.keyDownHandler(event)}
```

Comment:
I don't know if it's common, to set the keyDownListener to the hosting node, but I thought, especially when there will be more workspaceTypes perhaps in the future, we could save some locs this way.